pkg/operation/botanist/etcdencryption.go

```diff
@@ -0,0 +1,277 @@
+	err := b.writeEncryptionConfigurationSecretToSeed(ctx, ecYamlBytes, forcePlaintextSecrets)
+	if err != nil {
+		return err
+	}
+	return b.writeEncryptionConfigurationSecretToGarden(ctx, ecYamlBytes)
+}
+
+// createEncryptionConfigurationSecret creates the secret object containing the EncryptionConfiguration
+func (b *Botanist) createEncryptionConfigurationSecret(name string, namespace string, ecYamlBytes []byte, forcePlaintextSecrets bool) *corev1.Secret {
+	secretObj := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			common.EtcdEncryptionSecretFileName: ecYamlBytes,
+		},
+	}
+	if forcePlaintextSecrets {
+		secretObj.ObjectMeta.Annotations = map[string]string{
+			common.EtcdEncryptionForcePlaintextAnnotationName: "true",
+		}
+	}
+	return &secretObj
+}
+
+// writeEncryptionConfigurationSecretToSeed writes the secret which contains the EncryptionConfiguration
+// to the shoot namespace in the seed cluster
+func (b *Botanist) writeEncryptionConfigurationSecretToSeed(ctx context.Context, ecYamlBytes []byte, forcePlaintextSecrets bool) error {
+	secretObj := b.createEncryptionConfigurationSecret(common.EtcdEncryptionSecretName, b.Shoot.SeedNamespace, ecYamlBytes, forcePlaintextSecrets)
+	return kutil.CreateOrUpdate(ctx, b.K8sSeedClient.Client(), secretObj, func() error {
+		secretObj.Data[common.EtcdEncryptionSecretFileName] = ecYamlBytes
+		if forcePlaintextSecrets {
+			secretObj.Annotations[common.EtcdEncryptionForcePlaintextAnnotationName] = "true"
+		}
+		return nil
+	})
+}
+
+// writeEncryptionConfigurationSecretToGarden writes the secret which contains the EncryptionConfiguration
+// to the garden cluster
+func (b *Botanist) writeEncryptionConfigurationSecretToGarden(ctx context.Context, ecYamlBytes []byte) error {
+	secretNameInGardenCluster := generateGardenSecretName(b.Shoot.Info.Name, common.EtcdEncryptionSecretName)
+	secretObj := b.createEncryptionConfigurationSecret(secretNameInGardenCluster, b.Shoot.Info.Namespace, ecYamlBytes, false)
+	secretObj.ObjectMeta.OwnerReferences = []metav1.OwnerReference{
+		*metav1.NewControllerRef(b.Shoot.Info, gardenv1beta1.SchemeGroupVersion.WithKind("Shoot")),
+	}
+	return kutil.CreateOrUpdate(ctx, b.K8sGardenClient.Client(), secretObj, func() error {
+		secretObj.Data[common.EtcdEncryptionSecretFileName] = ecYamlBytes
+		return nil
+	})
+}
+
+// isEncryptionConfigurationConsistent checks whether the configuration is consistent in seed and garden
+func (b *Botanist) isEncryptionConfigurationConsistent(ecSeed *apiserverconfigv1.EncryptionConfiguration, ecGarden *apiserverconfigv1.EncryptionConfiguration) error {
+	if !apiequality.Semantic.DeepEqual(ecSeed, ecGarden) {
+		return fmt.Errorf("EncryptionConfiguration in seed cluster and garden cluster are not equal")
+	}
+	err := encryptionconfiguration.Validate(ecSeed)
```

Comment:
Checking whether the configuration is valid should've already happened before comparing the seed to the garden configuration. You should do this immediately after loading it and then this function can be omitted since it's only a equality check.